README.md

```diff
@@ -111,6 +111,39 @@ provided:
 See `roles/keycloak/defaults/main.yml` for a list of other variable
 defaults that one may want to override.
 
+Testing
+-------
+In-tree tests are provided that use molecule to test the role against docker containers.
+These tests are designed to be used by CI, but they can also be run locally to test it
+out while developing.  This is best done by installing molecule in a virtualenv:
+
+  `$ virtualenv .venv`
+  `$ source .venv/bin/activate`
+  `$ pip install molecule docker`
+
+It is required to run the tests as a user who is authorized to run the 'docker' command
+without using sudo.  This is typically accomplished by adding your user to the 'docker'
+group on your system.
+
+Additionally, there is a challenge around python-libselinux on platforms that use SELinux.
+If you are using a virtualenv, you need to make sure that the selinux python module is
+available in the virtualenv.  Even if it is installed on your ansible controller host
+and the target host, some of the tasks that are delegated to the locahost will use the
+virtualenv.  The selinux module can't be installed via pip.  A workaround for this is
+to copy the entire `selinux` directory from your system site-packages location into
+the virtualenv site-packages.  You also need to copy the `_selinux.so` file from
+site-locations as well.
```

Comment:
What would python-libselinux be needed for here?

Reply:
When ansible copies files or creates files from templates on a system with SELinux, it uses libselinux (likely for setting getting the fcontext for the task result).  If you want to read about the mess of this issue, this is a good summary:

  https://dmsimard.com/2016/01/08/selinux-python-virtualenv-chroot-and-ansible-dont-play-nice/

Reply:
It's used by ansible, see https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#managed-node-requirements

An alternative would be to create the `virtualenv` with `--system-site-packages`.

Reply:
Using `--system-site-packages` would be ideal, but I encountered dependency issues when using this approach on Fedora 28.  When attempting to `pip install molecule` in a `virtualenv` that uses `--system-site-packages`, I received the following error:

  `reno 2.0.3 has requirement pbr<2.0,>=1.4, but you'll have pbr 4.1.0 which is incompatible.`